crates/uv-resolver/src/resolution/graph.rs

```diff
@@ -32,6 +32,13 @@ pub struct ResolutionGraph {
     pub(crate) diagnostics: Vec<ResolutionDiagnostic>,
 }
 
+type NodeKey<'a> = (
+    &'a PackageName,
+    &'a Version,
+    Option<&'a ExtraName>,
+    Option<&'a ExtraName>,
```

Comment:
Nit: Can you add a comment which one is which

Reply:
I decided to introduce a separate type.